ConfigSpace/configuration_space.py

```diff
@@ -113,9 +116,10 @@ def add_hyperparameter(self, hyperparameter):
         self._check_default_configuration()
         self._sort_hyperparameters()
 
+        # todo: check it returns input as aoutput?
         return hyperparameter
 
-    def add_condition(self, condition):
+    def add_condition(self, condition: ConditionComponent) -> ConditionComponent:
```

Comment:
the first line of this function raises error if input is not of type conditioncomponent
        **if not isinstance(condition, ConditionComponent):
            raise TypeError("The method add_condition must be called "
                            "with an instance of "
                            "HPOlibConfigSpace.condition.ConditionComponent.")**